src/incubator/Slider/index.tsx

```diff
@@ -0,0 +1,390 @@
+
+  const defaultThumbStyle: StyleProp<ViewStyle> = [
+    styles.thumb,
+    {backgroundColor: disabled ? Colors.$backgroundDisabled : thumbTintColor || Colors.$backgroundPrimaryHeavy}
+  ];
+  const _thumbStyle = useSharedValue(JSON.parse(JSON.stringify(StyleSheet.flatten(thumbStyle || defaultThumbStyle))));
+  const _activeThumbStyle = 
+    useSharedValue(activeThumbStyle ? JSON.parse(JSON.stringify(StyleSheet.flatten(activeThumbStyle))) : undefined);
+
+  const trackAnimatedStyles = useAnimatedStyle(() => {
+    if (useRange) {
+      return {
+        transform: [{translateX: defaultThumbOffset.value.x * rtlFix}],
+        width: activeTrackWidth.value
+      };
+    } else {
+      return {
+        width: activeTrackWidth.value
+      };
+    }
+  });
+  
+  const defaultThumbAnimatedStyles = useAnimatedStyle(() => {
+    const activeStyle = isPressedDefault.value ? _activeThumbStyle.value : _thumbStyle.value;
+    return {
+      transform: [
+        {translateX: (defaultThumbOffset.value.x - thumbSize.value.width / 2) * rtlFix},
+        {scale: withSpring(!disableActiveStyling && isPressedDefault.value ? 1.3 : 1)}
+      ],
+      ...activeStyle
+    };
+  });
+
+  const rangeThumbAnimatedStyles = useAnimatedStyle(() => {
+    const activeStyle = isPressedDefault.value ? _activeThumbStyle.value : _thumbStyle.value;
+    return {
+      transform: [
+        {translateX: (rangeThumbOffset.value.x - thumbSize.value.width / 2) * rtlFix},
+        {scale: withSpring(!disableActiveStyling && isPressedRange.value ? 1.3 : 1)}
+      ],
+      ...activeStyle
+    };
+  });
+
+  const updateDefaultThumb = (x: number) => {
+    'worklet';
+    defaultThumbOffset.value = {x, y: 0};
+    defaultThumbStart.value = {x, y: 0};
+    activeTrackWidth.value = Math.abs(useRange ? rangeThumbStart.value.x - x : x);
+    runOnJS(onChange)(useRange ? {min: x, max: rangeThumbStart.value.x} : x);
+  };
+
+  const updateRangeThumb = (x: number) => {
+    'worklet';
+    rangeThumbOffset.value = {x, y: 0};
+    rangeThumbStart.value = {x, y: 0};
+    activeTrackWidth.value = x - defaultThumbStart.value.x;
+    runOnJS(onChange)({min: defaultThumbStart.value.x, max: x});
+  };
+
+  const defaultThumbGesture = Gesture.Pan()
+    .onBegin(() => {
+      isPressedDefault.value = true;
+      activeThumb.value = ThumbType.DEFAULT;
+    })
+    .onUpdate(e => {
+      onSeekStart?.();
+      let newX = defaultThumbStart.value.x + e.translationX * (shouldDisableRTL ? 1 : rtlFix);
+      if (newX < 0) { // bottom edge
+        newX = 0;
+      } else if (!useRange && newX > trackSize.value.width) { // top edge
+        newX = trackSize.value.width;
+      }
+      if (newX <= rangeThumbStart.value.x - rangeGap && newX >= 0) { // range
+        defaultThumbOffset.value = {x: newX, y: 0};
+        activeTrackWidth.value = useRange ? rangeThumbStart.value.x - newX : newX;
+        runOnJS(onChange)(useRange ? {min: newX, max: rangeThumbStart.value.x} : newX);
```

Comment:
Consider having a dedicate `useAnimatedReaction` that handles the data and the onChange callback
There you can normalize the (decimal) values so the onChange won't trigger that often. It might even fix the weird behavior you have now 

```
  useAnimatedReaction(() => {
    return Math.round(defaultThumbOffset.value);
  },
  (selected, previous) => {
     // Report onChange only when the value changes 
  },
```

Reply:
Why normalize the decimal values? The 'step' prop is the one to use if the user wishes to round the values... I've considered throttling the 'onChange' but that will make the slider inaccurate.